tests/nativeapp/test_event_sharing.py

```diff
@@ -0,0 +1,1086 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from textwrap import dedent
+from unittest import mock
+from unittest.mock import MagicMock
+
+import pytest
+from click import ClickException
+from snowflake.cli._plugins.connection.util import UIParameter
+from snowflake.cli._plugins.nativeapp.constants import (
+    SPECIAL_COMMENT,
+)
+from snowflake.cli._plugins.nativeapp.entities.application import (
+    ApplicationEntity,
+    ApplicationEntityModel,
+)
+from snowflake.cli._plugins.nativeapp.entities.application_package import (
+    ApplicationPackageEntity,
+    ApplicationPackageEntityModel,
+)
+from snowflake.cli._plugins.nativeapp.policy import (
+    AllowAlwaysPolicy,
+    AskAlwaysPolicy,
+    DenyAlwaysPolicy,
+    PolicyBase,
+)
+from snowflake.cli._plugins.nativeapp.same_account_install_method import (
+    SameAccountInstallMethod,
+)
+from snowflake.cli._plugins.stage.diff import DiffResult
+from snowflake.cli._plugins.workspace.context import ActionContext, WorkspaceContext
+from snowflake.cli.api.console import cli_console as cc
+from snowflake.cli.api.console.abc import AbstractConsole
+from snowflake.cli.api.project.definition_manager import DefinitionManager
+
+from tests.nativeapp.factories import (
+    ApplicationEntityModelFactory,
+    ApplicationPackageEntityModelFactory,
+    ProjectV2Factory,
+)
+from tests.nativeapp.patch_utils import (
+    mock_connection,
+)
+from tests.nativeapp.utils import (
+    APP_ENTITY_GET_EXISTING_APP_INFO,
+    GET_UI_PARAMETERS,
+    SQL_EXECUTOR_EXECUTE,
+    mock_execute_helper,
+)
+from tests.testing_utils.fixtures import MockConnectionCtx
+
+allow_always_policy = AllowAlwaysPolicy()
+ask_always_policy = AskAlwaysPolicy()
+deny_always_policy = DenyAlwaysPolicy()
```

Comment:
these are unused in this file. your `_create_or_upgrade_app` takes in a `policy` param, did you mean to use them there?

Reply:
no it's a left-over copy from another file.